File: Domain/Entities/Match.cs
using Infra.Interface;
using Infra.enums;
using System.ComponentModel.DataAnnotations.Schema;

namespace Data.Entities
{
    public class Match : ISoftDeletableEntity
    {
        public Guid Id { get; set; }

        [ForeignKey(nameof(HomeTeam))]
        public Guid? HomeTeamId { get; set; }
        public Team? HomeTeam { get; set; }

        [ForeignKey(nameof(AwayTeam))]
        public Guid? AwayTeamId { get; set; }
        public Team? AwayTeam { get; set; }

        public string? HomeTeamPlaceholder { get; set; }
        public string? AwayTeamPlaceholder { get; set; }

        public int RoundNumber { get; set; }

        public DateTime KickoffTime { get; set; }
        public DateTime? FinalWhistleTime { get; set; }

        [ForeignKey(nameof(Tournament))]
        public Guid TournamentId { get; set; }
        public Tournament Tournament { get; set; } = null!;

        [ForeignKey(nameof(Group))]
        public Guid? GroupId { get; set; }
        public Group? Group { get; set; }
        public MatchStatus Status { get; set; }
        public string? Venue { get; set; }
        public StageType StageType { get; set; }

        public ICollection<MatchCard> Cards { get; set; } = [];
        public ICollection<MatchGoal> Goals { get; set; } = [];
        public bool IsDeleted { get; set; }
        public DateTimeOffset CreatedAt { get; set; }
        public DateTimeOffset? UpdatedAt { get; set; }
    }
}

File: soccer-frontend/src/app/core/models/match.ts
import { MatchStatus } from './enums';

export interface Match {
  id: string;
  groupId: string;
  homeTeamId: string;
  awayTeamId: string;
  homeTeamName?: string;
  awayTeamName?: string;
  homeTeamGoals: number;
  awayTeamGoals: number;
  status: MatchStatus;
  matchDate: Date;
  createdAt: Date;
  updatedAt: Date;
}

export interface MatchDetail extends Match {
  goals: MatchGoal[];
  cards: MatchCard[];
  lineups: MatchLineup[];
}

export interface MatchLineup {
  id: string;
  matchId: string;
  teamId: string;
  playerId: string;
  playerName?: string;
  jerseyNumber?: number;
  position: string;
  isStarting: boolean;
  createdAt: Date;
}

export interface MatchGoal {
  id: string;
  matchId: string;
  teamId: string;
  playerId: string;
  playerName?: string;
  minute: number;
  type: string;
  createdAt: Date;
}

export interface MatchCard {
  id: string;
  matchId: string;
  teamId: string;
  playerId: string;
  playerName?: string;
  cardType: string;
  minute: number;
  createdAt: Date;
}
